packages/token/src/erc20/extensions/erc20_wrapper.cairo

```diff
@@ -0,0 +1,137 @@
+///
+/// Users can deposit and withdraw "underlying tokens" and receive a matching number of "wrapped
+/// tokens". This is useful in conjunction with other modules. For example, combining this wrapping
+/// mechanism with the ERC20VotesComponent will allow the wrapping of an existing "basic" ERC-20
+/// into a governance token.
+///
+/// WARNING: Any mechanism in which the underlying token changes the `balanceOf` of an account
+/// without an explicit transfer may desynchronize this contract's supply and its underlying
+/// balance. Please exercise caution when wrapping tokens that may undercollateralize the wrapper
+/// (i.e. the wrapper's total supply is higher than its underlying balance). See `recover` for
+/// recovering value accrued to the wrapper.
+#[starknet::component]
+pub mod ERC20WrapperComponent {
+    use core::num::traits::Zero;
+    use openzeppelin_interfaces::erc20::{IERC20Dispatcher, IERC20DispatcherTrait};
+    use openzeppelin_interfaces::token::erc20::{IERC20, IERC20Wrapper};
+    use starknet::ContractAddress;
+    use starknet::storage::{StoragePointerReadAccess, StoragePointerWriteAccess};
+    use crate::erc20::ERC20Component;
+    use crate::erc20::ERC20Component::InternalImpl as ERC20InternalImpl;
+
+    #[storage]
+    pub struct Storage {
+        pub ERC20Wrapper_underlying: ContractAddress,
+    }
+
+    pub mod Errors {
+        pub const INVALID_UNDERLYING_ADDRESS: felt252 = 'Wrapper: invalid underlying';
+        pub const INVALID_SENDER: felt252 = 'Wrapper: invalid sender';
+        pub const INVALID_RECEIVER: felt252 = 'Wrapper: invalid receiver';
+        pub const TRANSFER_FAILED: felt252 = 'Wrapper: transfer failed';
+    }
+
+    //
+    // External
+    //
+
+    #[embeddable_as(ERC20WrapperImpl)]
+    impl ERC20Wrapper<
+        TContractState,
+        +HasComponent<TContractState>,
+        impl ERC20: ERC20Component::HasComponent<TContractState>,
+        +ERC20Component::ERC20HooksTrait<TContractState>,
+        +Drop<TContractState>,
+    > of IERC20Wrapper<ComponentState<TContractState>> {
+        /// Returns the address of the underlying token used for the wrapper.
+        fn underlying(self: @ComponentState<TContractState>) -> ContractAddress {
+            self.ERC20Wrapper_underlying.read()
+        }
+
+        /// Deposits underlying tokens and mints the same amount of wrapped tokens to `receiver`.
+        fn deposit_for(
+            ref self: ComponentState<TContractState>, receiver: ContractAddress, amount: u256,
+        ) -> bool {
+            let caller = starknet::get_caller_address();
+            let this = starknet::get_contract_address();
+            assert(caller != this, Errors::INVALID_SENDER);
+            assert(receiver != this, Errors::INVALID_RECEIVER);
+
+            let underlying = self.underlying();
+            let token = IERC20Dispatcher { contract_address: underlying };
+            let ok = token.transfer_from(caller, this, amount);
```

Comment:
Based on [Solidity reference](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/extensions/ERC20Wrapper.sol#L63) it should use `SafeErc20` utility.